src/main/java/com/iota/iri/service/API.java

```diff
@@ -737,16 +964,26 @@ private AbstractResponse getNodeInfoStatement(){
      * This is for determining if a transaction was accepted and confirmed by the network or not.
      * You can search for multiple tips (and thus, milestones) to get past inclusion states of transactions.
      *
-     * This API call simply returns a list of boolean values in the same order as the transaction list you submitted, thus you get a true/false whether a transaction is confirmed or not.
+     * This API call returns a list of boolean values in the same order as the submitted transaction 
+     * Booleans will be <code>true</code> if a transaction is confirmed, otherwise <code>false</code>.
      * Returns an {@link com.iota.iri.service.dto.ErrorResponse} if a tip is missing or the subtangle is not solid
      *
      * @param transactions List of transactions you want to get the inclusion state for.
      * @param tips List of tips (including milestones) you want to search for the inclusion state.
      * @return {@link com.iota.iri.service.dto.GetInclusionStatesResponse}
+     * @throws Exception When a transaction cannot be loaded from hash
      **/
-    private AbstractResponse getInclusionStatesStatement(final List<String> transactions, final List<String> tips) throws Exception {
-        final List<Hash> trans = transactions.stream().map(HashFactory.TRANSACTION::create).collect(Collectors.toList());
-        final List<Hash> tps = tips.stream().map(HashFactory.TRANSACTION::create).collect(Collectors.toList());
+    private AbstractResponse getInclusionStatesStatement(
```

Comment:
it seems to me that comments should be at a higher level, to convey the intent/algorithm used.
most of the lines "just" describe what the code does, not why/what it means.

for example:
```
Sets to -1 if is only known by hash, or has no index, or index is above the max tip index. (not included)
```
the intent is: set to -1 if the transaction is not referenced by any of the tips. (because the tx is either above all tips - higher index, or unconfirmed index 0)